src/js_native_api_v8.cc

```diff
@@ -870,7 +870,14 @@ napi_status napi_get_property_names(napi_env env,
   v8::Local<v8::Object> obj;
   CHECK_TO_OBJECT(env, context, obj, object);
 
-  auto maybe_propertynames = obj->GetPropertyNames(context);
+  v8::MaybeLocal<v8::Array> maybe_propertynames = obj->GetPropertyNames(
+    context,
+    v8::KeyCollectionMode::kIncludePrototypes,
```

Comment:
Should `napi_get_property_names` return all property names including these on prototypes? Or `kOwnOnly` is just fine.

Reply:
I think that would have been a good idea when this API was originally conceived, but it would be a breaking change to switch this over, and we generally don’t do that for N-API.

Reply:
I think it's a good idea to add another `napi_get_own_propery_names`. I can do that if others agree.

Reply:
@bridgear I think that’s a good idea. I think we’d also maybe want to include non-enumerable properties/symbols, because there doesn’t currently appear to be a way to get those (maybe as an option, similar to the way in which the V8 API leaves us the choice).